### ISSUE TITLE
[rule: no-shadow] false positive caused by a path string

### ISSUE THREAD

**Issue by lucendio** [eslint/eslint#5450]
**What version of ESLint are you using?**
2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Espree

**What did you do? Please include the actual source code causing the issue.**

ESLint config excerpt:
```yaml
no-shadow: [
    2,
    {
        builtinGlobals: true,
        hoist: 'functions',
        allow: []
    }
]
```
js file content:
```js
const path = '/account/:name';
let name = 'unimportant value';
```

**What did you expect to happen?**
variable declaration of ``name``(line:2) doesnt get marked as a problem

**What actually happened? Please include the actual, raw output from ESLint.**
variable declaration of ``name``(line:2) gets marked as a problem

**output message (within webstorm)**
``ESLint: 'name' is already declared in the upper scope. (no-shadow)``

 My guess: regex for finding a previous usage needs a bit more work

**Comment by gyandeeps** [comment]
Can you please share the exact output?

>What did you expect to happen?
variable declaration of name(line:2) doesnt get marked as a problem

>What actually happened? Please include the actual, raw output from ESLint.
variable declaration of name(line:2) gets marked as a problem

Currently your expect vs happened response seems to contradict each other.

**Comment by lucendio** [comment]
I am sorry. my mistake. I added the output mesage

**Comment by platinumazure** [comment]
You should show your whole config- it's possible an environment is enabling a global called name.

**Comment by lucendio** [comment]
[eslintrc.txt](https://github.com/eslint/eslint/files/153847/eslintrc.txt)
NOTE: its a ``.yml`` file, but unfortunately github does not support uploading such extension. so I had to rename it.
I thin you might be right, because renaming both, ``:name`` string (line:1) and variable name preventing the error

**Comment by mysticatea** [comment]
The `name` variable is defined in the global of `browser` environment.
So this is correct behavior.

**Comment by faceleg** [comment]
Closing as not a bug.

**Comment by lucendio** [comment]
thanks for clearing this up. I am sorry for causing extra work.

**Comment by faceleg** [comment]
No problem, there are a few of us here whose role is helping manage issues :)
